cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkTransactional.java

```diff
@@ -133,6 +133,11 @@ public void execute(TxRunnable runnable) throws TransactionFailureException {
     execute(wrap(runnable), TransactionType.EXPLICIT);
   }
 
+  @Override
+  public void execute(int timeout, TxRunnable runnable) throws TransactionFailureException {
+    throw new TransactionFailureException("Transaction with explicit timeout is not supported in Spark");
```

Comment:
It might be good to have the trait `SparkExecutionContext` and the class `JavaSparkExecutionContext` overrides the `execute(timeout, runnable)` method hence override the javadoc to state clearly that execute with timeout is not supported in Spark. It helps showing the correct javadoc in IDE.